### PR TITLE
Add a Universal Ornaments section to Records

### PR DESCRIPTION
**Pull request by robojumper**
People have been asking for Collections to indicate universal ornament unlock status. However, collections/collectibles do not actually reflect ornaments completely -- some ornaments do not have collections entries and in some other cases multiple ornament sets map to the same collections node (more or less). So this simply adds another Records section, entirely driven by the universal ornament plugSet data, and tries to organize these plugs according to heuristics to attach them to collections presentation nodes. This works really well with many sets (note e.g. the two Solstice sets where one set has a glow and the other doesn't):

grafik

For some ornaments, this falls short -- we'll likely need to generate about 6 sets in D2AI

grafik

And Bungie also messed up the Solstice set for Hunter, so that makes two more sets in D2AI

grafik

but otherwise this works without much hardcoding

### CODE REVIEW

**Comment by nev-r** [comment]
any chance we could live-match icon urls, to solve more of the weird cases?

**Comment by robojumper** [comment]
I don't quite understand what you mean by "live-match icon urls" and which "weird cases" you're referring to -- the two different Titan sets are working as intended, as they're different sets. The problem with all these pieces in "Others" is that they literally aren't in collections at all, and I don't think there's a good way to assign them to their own armor sets without that info.

**Comment by bhollis** [comment]
Seems to work pretty well! I like that I can see the things I have unlocked in collections but which I haven't unlocked as ornaments.

File: config/i18n.json [hunk 1]
```diff
@@ -998,7 +998,9 @@
     "RecordBooks": "Record Books"
   },
   "Records": {
-    "Title": "Records"
+    "Title": "Records",
+    "UniversalOrnaments": "Universal Ornaments",
```
Comment (nev-r) [inline config/i18n.json:1002]: 3655910122 2497860177 or 466446949 have a pre-localized and bungie-standard copy of this string

**Comment by bhollis** [comment]
This seems pretty cool - what's left to do on it?

**Comment by robojumper** [comment]
> What's left to do on it?

I wasn't quite certain if this is something DIM should have. If we want this, we need https://github.com/DestinyItemManager/d2-additional-info/pull/494 merged and this needs one more commit to organize the remaining armor sets using that info.

**Comment by bhollis** [comment]
I think it's not a bad fit for the Records page. I don't want to get too far into curated data but this is mostly still from the definitions.

**Review by bhollis — APPROVED** [review]
Looks great!

**Pull request merged**